### PR TITLE
Fix attrs pos

### LINKED ISSUE #60042: [doc] Unwanted padding in type declaration
You can see it in this type declaration: https://doc.rust-lang.org/nightly/nightly-rustc/rustc/mir/enum.PlaceBase.html

Screenshot:

Screenshot 2019-05-12 at 14 45 56

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Fixes #60042.

Screenshot:

Screenshot 2019-05-12 at 15 02 25

r? @rust-lang/rustdoc

### CODE REVIEW

**Comment by GuillaumeGomez** [comment]
ping

**Comment by Manishearth** [comment]
I don't understand what this is fixing, and I don't understand how. When filing such issues please include screenshots, and include a before/after in the PR.

Also you've added some parameters to some functions here but it's totally unclear what they do, please document them.

**Comment by GuillaumeGomez** [comment]
I thought the display on the issue was clear enough, my bad... The point here is to fix the alignment of the `#[...]` attributes on the types declaration. Those new arguments allow to add a new class for attributes generated inside the type declaration in order to prevent those alignment issues. The screenshot itself would just be a "normal" looking type declaration, which doesn't really makes sense on its own since it's supposed to look like this from the beginning.

**Comment by Manishearth** [comment]
There's no attribute there. I see padding, but no attribute.

A link to a broken page isn't good enough for this, because that page may not render the same for everyone, and once the bug is fixed it will be hard to know what was broken. Issues should be self-contained and make sense even after being fixed. So should pull requests, the justification for this should be in the comments.

The screenshot in a PR helps highlight what was changed by focusing on that. It's less necessary since usually it's just normal, but the screenshot in the issue is pretty important.


image

**Comment by GuillaumeGomez** [comment]
This one is already present in the current page. Also it's a broken link, not a screenshot?! Damn it... I'll try to change it later on. Sorry about the mess...

**Comment by GuillaumeGomez** [comment]
@Manishearth I added a screenshot on both the issue and the PR.

File: src/librustdoc/html/render.rs [hunk 1]
```diff
@@ -3783,7 +3789,7 @@ const ATTRIBUTE_WHITELIST: &'static [&'static str] = &[
     "non_exhaustive"
 ];
 
-fn render_attributes(w: &mut dyn fmt::Write, it: &clean::Item) -> fmt::Result {
+fn render_attributes(w: &mut dyn fmt::Write, it: &clean::Item, top: bool) -> fmt::Result {
```
Comment (Manishearth) [inline src/librustdoc/html/render.rs]: Still need comments explaining what this parameter does.

**Review by Manishearth — APPROVED** [review]
code looks good, please add some comments on the function and on the CSS

**Comment by GuillaumeGomez** [comment]
Yes sir!

**Comment by GuillaumeGomez** [comment]
Updated!

**Comment by GuillaumeGomez** [comment]
@Manishearth Is my comment good enough or do you have something else in mind?

**Pull request merged**